clang/runtime/dpct-rt/include/dpct/mma_utils.hpp

```diff
@@ -0,0 +1,117 @@
+//==---- mma_utils.hpp ----------------------------*- C++ -*----------------==//
+//
+// Copyright (C) Intel Corporation
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __DPCT_MMA_UTILS_HPP__
+#define __DPCT_MMA_UTILS_HPP__
+
+#include <sycl/sycl.hpp>
+
+namespace dpct {
+namespace experimental {
+namespace matrix {
+
+namespace sycl_matrix = sycl::ext::oneapi::experimental::matrix;
+
+struct row_major
+    : public std::integral_constant<sycl_matrix::layout,
+                                    sycl_matrix::layout::row_major> {};
+struct col_major
+    : public std::integral_constant<sycl_matrix::layout,
+                                    sycl_matrix::layout::col_major> {};
+struct a
+    : public std::integral_constant<sycl_matrix::use, sycl_matrix::use::a> {};
+struct b
+    : public std::integral_constant<sycl_matrix::use, sycl_matrix::use::b> {};
+struct accumulator
+    : public std::integral_constant<sycl_matrix::use,
+                                    sycl_matrix::use::accumulator> {};
+enum layout_t { m_row_major, m_col_major };
+
+template <typename use, int m, int n, int k, typename T,
+          typename layout = std::integral_constant<
+              sycl_matrix::layout, sycl_matrix::layout::dynamic>>
+class joint_matrix {
+  static constexpr int rows = (use::value == sycl_matrix::use::b) ? n : m;
```

Comment:
How about adding a new struct like:
```
template <class use, int m, int n, int k> struct matrix_traits;
template <int m, int n, int k> struct matrix_traits<...::a, m, n, k> {
 static constexpr int rows = m;
 static constexpr int cols = m;
};
```
It could be more readable.